models/curvefi/ethereum/curvefi_ethereum_view_pools.sql

```diff
@@ -0,0 +1,285 @@
+        deposit_contract AS deposit_contract,
+        gauge_contract AS gauge_contract,
+        coin0 AS coin0,
+        coin1 AS coin1,
+        coin2 AS coin2,
+        coin3 AS coin3,
+        undercoin0 AS undercoin0,
+        undercoin1 AS undercoin1,
+        undercoin2 AS undercoin2,
+        undercoin3 AS undercoin3
+    FROM
+        records
+),
+regular_pools_deployed AS (
+    SELECT
+        version,
+        `name`,
+        symbol,
+        pool_address,
+        CAST(
+            NULL AS DECIMAL
+        ) AS A,
+        CAST(
+            NULL AS DECIMAL
+        ) AS mid_fee,
+        CAST(
+            NULL AS DECIMAL
+        ) AS out_fee,
+        token_address,
+        deposit_contract,
+        coin0,
+        coin1,
+        coin2,
+        coin3,
+        undercoin0,
+        undercoin1,
+        undercoin2,
+        undercoin3,
+        gauge_contract
+    FROM
+        regular_pools
+),
+---------------------------------------------------------------- V1 Pools ----------------------------------------------------------------
+plain_calls AS (
+    SELECT
+        _name,
+        _symbol,
+        output_0,
+        _coins,
+        `_A`,
+        _fee
+    FROM
+        {{ source(
+            'curvefi_ethereum',
+            'CurveFactory_call_deploy_plain_pool'
+        ) }}
+    WHERE
+        call_success
+),
+plain_pools_deployed AS (
+    SELECT
+        'Factory V1 Plain' AS version,
+        _name AS `name`,
+        _symbol AS symbol,
+        output_0 AS pool_address,
+        "_A" AS A,
+        _fee AS mid_fee,
+        _fee AS out_fee,
+        output_0 AS token_address,
+        output_0 AS deposit_contract,
+        _coins [1] AS coin0,
+        _coins [2] AS coin1,
+        _coins [3] AS coin2,
+        _coins [4] AS coin3,
+        CAST(
+            NULL AS STRING
+        ) AS undercoin0,
+        CAST(
+            NULL AS STRING
+        ) AS undercoin1,
+        CAST(
+            NULL AS STRING
+        ) AS undercoin2,
+        CAST(
+            NULL AS STRING
+        ) AS undercoin3
+    FROM
+        plain_calls
+),
+meta_calls AS (
+    SELECT
+        _name,
+        _symbol,
+        output_0,
+        call_tx_hash,
+        _base_pool,
+        _coin,
+        "_A",
+        _fee
+    FROM
+        {{ source(
+            'curvefi_ethereum',
+            'CurveFactory_call_deploy_metapool'
+        ) }}
+    WHERE
+        call_success
+),
+meta_pools_deployed AS (
+    SELECT
+        'Factory V1 Meta' AS version,
+        _name AS `name`,
+        _symbol AS symbol,
+        output_0 AS pool_address,
+        "_A" AS A,
+        _fee AS mid_fee,
+        _fee AS out_fee,
+        output_0 AS token_address,
+        output_0 AS deposit_contract,
+        _coin AS coin0,
+        CASE
+            WHEN _base_pool = '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7' THEN '0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490' --changing from swap to token contract
```

Comment:
I think setting these as variables would be a really good showcase of best practices for the wizards.

Reply:
converted to variables! good call!